File: AlgorithmsPackCPP/Minimum Operations to Convert Number.cpp
class Solution {
public:
    bool process(queue<int>&q, vector<bool>&visited, int num, int goal){
        if(num==goal)
            return true;
        
        if(0<=num && num<=1000 && !visited[num]){
            visited[num] = true;
            q.push(num);
        }
        
        return false;
    }
    
    int minimumOperations(vector<int>& nums, int start, int goal) {
        queue<int>q;
        vector<bool>visited(1001);
        q.push(start);
        visited[start] = true;
        
        for(int level = 0, sz = q.size(); !q.empty(); ++level, sz = q.size()){
            while(sz--){
                int cur = q.front();
                q.pop();
                
                for(int i = 0; i < nums.size(); i++){
                    if(process(q,visited,cur+nums[i],goal)
                      ||process(q,visited,cur-nums[i],goal)
                      ||process(q,visited,cur^nums[i],goal))
                        return level+1;
                }
            }
        }
        return -1;
    }
};

File: AlgorithmsPackPythonSolutions/Minimum Operations to Convert Number.py
class Solution:
    def process(self, q, visited, num, goal):
        if num == goal:
            return True
        
        if 0 <= num <= 1000 and not visited[num]:
            q.append(num)
            visited[num] = True
        
        return False
    
    def minimumOperations(self, nums: List[int], start: int, goal: int) -> int:
        q = collections.deque()
        visited = [False] * 1001
        q.append(start)
        visited[start] = True
        
        level = 0
        while q:
            sz = len(q)
            for _ in range(sz):
                x = q.popleft()
                for num in nums:
                    if (self.process(q, visited, x + num, goal) or
                        self.process(q, visited, x - num, goal) or
                        self.process(q, visited, x ^ num, goal)):
                        return level + 1
            level += 1
        
        return -1
